File: 프로그래머스/unrated/132267. 콜라 문제/콜라 문제.js
function solution(a, b, n) {
    // 리턴할 변수
    let result = 0;

    while (n >= a) {
        const cola = Math.floor(n / a) * b; // 콜라를 받음
        result += cola;

        // 남은 빈 병과 새로 받은 콜라를 더함
        n = cola + (n % a);
    }

    return result;
}

File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.java
class Solution {
    public int solution(int a, int b, int n) {
        int result = 0;
        
        while (n >= a) {
            int trade = n / a;
            int remain = n % a;
            
            result += trade * b;
            n = trade * b + remain;
        }
        
        return result;
    }
}
